### ISSUE TITLE
Support for Polymorphism

### ISSUE THREAD

**Issue by mpluecker** [BraintagsGmbH/vertx-pojo-mapper#59]
We want to be able to persist implementations of an abstract class in the same table / collection. In the following example we have an abstract class Animal with its implementations Dog and Fish. All dogs and fishes should be persisted in the collection “Animal”. When querying the animal collection, we want to get objects of the correct implementation class. Thus, the class of the implementation has to be stored with the serialized data. A similar annotation like @JsonTypeInfo would a way to configure this.

@Entity(name = "Animal")
public abstract class Animal {
  @Id
  public String id = null;
}
public class Dog extends Animal {
  public String favouriteFood = "Dog's Favourite";
}
public class Fish extends Animal {
  public float swimSpeed = 1.5f;
}

**Comment by remi128** [comment]
http://stackoverflow.com/questions/30362446/deserialize-json-with-jackson-into-polymorphic-types-a-complete-example-is-giv

**Comment by remi128** [comment]
implemented in branch #76-JMF, merging outstanding; annotation Entity got new property polyClass, which must define the common superclass or interface of all participant classes on a polymorphy. Additionally the already existing property Entity.name must be used to define the common destination collection / table

In the upper example this would be:

```
@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, 
      include = JsonTypeInfo.As.PROPERTY, property = "@class")
public abstract class Animal {
  @Id
  public String id = null;
  }
}

@Entity(name = "Animal", polyClass = Animal.class)
public class Dog extends Animal {
  public String favouriteFood = "Dog's Favourite";
}

@Entity(name = "Animal", polyClass = Animal.class)
public class Fish extends Animal {
  public float swimSpeed = 1.5f;
}
```

**Comment by remi128** [comment]
merged into develop
